geomstats/geometry/fiber_bundle.py

```diff
@@ -222,6 +222,12 @@ def tangent_riemannian_submersion(self, tangent_vec, base_point):
         projection: array-like, shape=[..., {base_dim, [n, m]}]
             Tangent vector to the base manifold.
         """
+        caller_name = sys._getframe().f_back.f_code.co_name
```

Comment:
Why do we need a syntax like this? Why not a NotImplementedError in the parent class, that does not appear when the class is indeed overridden?

The current syntax with == "horizontal_projection" does not seem robust to changes of names in methods, so I'm not sure it's a good practice?

Reply:
Definitely not a good practice, though it is hard to find a better alternative that is as general.

No problems with maintaining this in any case, as our tests will scream at us if we change these names (also, these names are quite stable, we're probably not changing them at any time soon).

Reply:
Above comment is already updated. New solution seems to be more clear.